python/cudf/cudf/core/buffer.py

```diff
@@ -102,8 +108,8 @@ def _init_from_array_like(self, data, owner):
                 f"Cannot construct Buffer from {data.__class__.__name__}"
             )
 
-    def serialize(self):
-        header = {}
+    def serialize(self) -> Tuple[dict, list]:
+        header = {}  # type: Dict[Any, Any]
```

Comment:
There's no way for MyPy to infer the type of `header` here. For example, it could as well be a `Dict[Any, int]`. See [here](https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html#explicit-types-for-collections) for a nice explanation.